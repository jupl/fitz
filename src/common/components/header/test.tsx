import {create} from 'react-test-renderer'

test('render', async () => {
  const React = await import('react')
  const {Header} = await import('.')
  const component = create(<Header title="Title" />)
  expect(component.toJSON()).toMatchSnapshot()
})
