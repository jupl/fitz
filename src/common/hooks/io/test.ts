import {expect, test} from '@jest/globals'
import {renderHook} from '@testing-library/react-hooks'

test('useDecode hook', async () => {
  const IO = await import('io-ts')
  const {useDecode} = await import('.')
  const {result} = renderHook(() => useDecode(IO.number, 1))
  expect(result.current).toMatchSnapshot()
})

test('useEncode hook', async () => {
  const IO = await import('io-ts')
  const {useEncode} = await import('.')
  const {result} = renderHook(() => useEncode(IO.number, 1))
  expect(result.current).toMatchSnapshot()
})
