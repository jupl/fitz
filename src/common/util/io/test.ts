test('decode', async () => {
  const IO = await import('io-ts')
  const {decode} = await import('.')
  expect(decode(IO.number, 1)).toBe(1)
})

test('decode with fallback', async () => {
  const IO = await import('io-ts')
  const {decode} = await import('.')
  expect(decode(IO.number, 'a', 1)).toBe(1)
})

test('decode with error', async () => {
  const IO = await import('io-ts')
  const {decode} = await import('.')
  expect(() => decode(IO.number, 'a')).toThrow(
    'Expecting number but instead got: "a"',
  )
})
