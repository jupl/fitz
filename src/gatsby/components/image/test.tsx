import {create} from 'react-test-renderer'

test('render with fixed Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    childImageSharp: {
      fixed: {
        height: 100,
        src: '/some/url',
        srcSet: '/some/url 100w, /another/url 200w',
        width: 100,
      },
    },
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('render with fluid Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        presentationWidth: 100,
        sizes: '(max-width: 50px) 50vw, 100px',
        src: '/some/url',
        srcSet: '/some/url 100w, /another/url 200w',
      },
    },
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})

test('render with no Sharp image', async () => {
  const React = await import('react')
  const {GatsbyImage} = await import('.')
  const data = {
    url: '/some/url',
  }
  const component = create(<GatsbyImage data={data} />)
  expect(component.toJSON()).toMatchSnapshot()
})
