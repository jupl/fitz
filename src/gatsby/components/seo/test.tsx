import {afterAll, beforeAll, expect, jest, test} from '@jest/globals'
import {create} from 'react-test-renderer'

void beforeAll(async () => {
  const hooks = await import('../../hooks/metadata')
  jest.spyOn(hooks, 'useGatsbyMetadata').mockReturnValue({
    author: 'Author',
    description: 'Description',
    title: 'Title',
  })
})

void afterAll(async () => {
  const hooks = await import('../../hooks/metadata')
  jest.spyOn(hooks, 'useGatsbyMetadata').mockRestore()
})

test('render', async () => {
  const React = await import('react')
  const {Helmet} = await import('react-helmet')
  const {GatsbySEO} = await import('.')
  create(<GatsbySEO />)
  expect(Helmet.peek()).toMatchSnapshot()
})
