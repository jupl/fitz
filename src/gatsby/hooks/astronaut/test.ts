import {renderHook} from '@testing-library/react-hooks'

beforeAll(async () => {
  const gatsby = await import('gatsby')
  jest
    .spyOn(gatsby, 'useStaticQuery')
    .mockReturnValue({astronaut: {url: 'astronaut.svg'}})
})

afterAll(async () => {
  const gatsby = await import('gatsby')
  jest.spyOn(gatsby, 'useStaticQuery').mockRestore()
})

test('hook', async () => {
  const {useGatsbyAstronaut} = await import('.')
  const {result} = renderHook(() => useGatsbyAstronaut())
  expect(result.current).toMatchSnapshot()
})
