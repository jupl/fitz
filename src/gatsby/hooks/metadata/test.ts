import {renderHook} from '@testing-library/react-hooks'

beforeAll(async () => {
  const gatsby = await import('gatsby')
  jest.spyOn(gatsby, 'useStaticQuery').mockReturnValue({
    site: {
      metadata: {
        author: 'Author',
        description: 'Description',
        title: 'Title',
      },
    },
  })
})

afterAll(async () => {
  const gatsby = await import('gatsby')
  jest.spyOn(gatsby, 'useStaticQuery').mockRestore()
})

test('hook', async () => {
  const {useGatsbyMetadata} = await import('.')
  const {result} = renderHook(() => useGatsbyMetadata())
  expect(result.current).toMatchSnapshot()
})
