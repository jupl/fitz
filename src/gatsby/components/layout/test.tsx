import test from 'ava'
import React from 'react'
import {create} from 'react-test-renderer'
import {replace} from 'testdouble'
import {useFakeTimers} from 'sinon'

test.before(() => {
  replace('../../hooks/metadata', {
    useGatsbyMetadata: () => ({
      author: 'Author',
      description: 'Description',
      title: 'Title',
    }),
  })
  useFakeTimers({
    now: 1594385918516,
  })
})

test('render', async t => {
  const {GatsbyLayout} = await import('.')
  const component = create(<GatsbyLayout>Hello</GatsbyLayout>)
  t.snapshot(component.toJSON())
})
