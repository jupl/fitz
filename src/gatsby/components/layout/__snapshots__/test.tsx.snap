// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
Array [
  <header
    className=""
  >
    <div
      className=""
    >
      <h1>
        <a
          className=""
          href="/"
        >
          Title
        </a>
      </h1>
    </div>
  </header>,
  <div
    className=""
  >
    Hello
  </div>,
  <footer
    className=""
  >
    Â© 
    2020
    , Built with 
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
  </footer>,
]
`;
